### ISSUE TITLE
Generic exception tuple detected as not valid

### ISSUE THREAD

**Issue by alexandermalyga** [microsoft/pyright#6944]
When using a generic function that receives a tuple of Exception types, it's detected as invalid to use in the `except` block. This does not happen in a non-generic function:

```python
# No errors
def foo(*errors: type[BaseException]) -> int | BaseException:
    try:
        return 1
    except errors as e:
        return e


# error: "tuple[type[T@bar], ...]" is not a valid exception class
def bar[T: BaseException](*errors: type[T]) -> int | T:
    try:
        return 1
    except errors as e:
        return e
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.346, which I just published. It will also be included in a future release of pylance.
